packages/garbo/src/tasks/freeFight.ts

```diff
@@ -0,0 +1,612 @@
+import { CombatStrategy, Quest } from "grimoire-kolmafia";
+import {
+  availableAmount,
+  canEquip,
+  guildStoreAvailable,
+  handlingChoice,
+  Item,
+  itemAmount,
+  itemDropsArray,
+  itemType,
+  mallPrice,
+  Monster,
+  myClass,
+  myMaxhp,
+  mySoulsauce,
+  restoreHp,
+  retrieveItem,
+  runChoice,
+  runCombat,
+  Skill,
+  use,
+  useSkill,
+  visitUrl,
+  weaponHands,
+} from "kolmafia";
+import {
+  $class,
+  $effect,
+  $familiar,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $phyla,
+  $skill,
+  ChateauMantegna,
+  clamp,
+  CombatLoversLocket,
+  Delayed,
+  ensureEffect,
+  get,
+  have,
+  maxBy,
+  sum,
+  TunnelOfLove,
+  undelay,
+  uneffect,
+  Witchess,
+} from "libram";
+import { acquire } from "../acquire";
+import { Macro } from "../combat";
+import { globalOptions } from "../config";
+import { garboValue } from "../garboValue";
+import { freeFightOutfit } from "../outfit";
+import { GarboTask } from "./engine";
+
+type GarboFreeFightTask = GarboTask & {
+  combatCount: () => number;
+  tentacle: boolean; // if a tentacle fight can follow
+};
+
+const DEFAULT_FREE_FIGHT_TASK = {
+  // GarboTask
+  combat: new CombatStrategy().autoattack(Macro.basicCombat()),
```

Comment:
We have a `GarboStrategy` class that does a lot of the autoattack and macro handling for us; you can just do `new GarboStrategy(Macro.basicCombat())`